pkg/controller/cassandradatacenter/controller/configmap.go

```diff
@@ -157,14 +121,90 @@ func addCassandraYamlOverrides(config configurationResources) error {
 	})
 
 	if err != nil {
-		return err
+		// we're serializing a known structure to YAML -- if that fails...
+		panic(err)
 	}
 
-	configMapVolumeAddTextFile(config.configMap, config.volumeSource, "cassandra.yaml.d/001-operator-overrides.yaml", string(data))
+	addFileFn("cassandra.yaml.d/001-operator-overrides.yaml", string(data))
+}
+
+
+
+func addCassandraJVMOptions(cdc *cassandraoperatorv1alpha1.CassandraDataCenter, addFileFn func (path string, data string)) {
+	// TODO: should this be Limits or Requests?
+	memoryLimit := cdc.Spec.Resources.Limits.Memory().Value()
+
+	jvmHeapSize := maxInt64(minInt64(memoryLimit/2, GIBIBYTE), minInt64(memoryLimit/4, 8*GIBIBYTE))
+
+	youngGenSize := youngGen(jvmHeapSize)
+
+	useG1GC := jvmHeapSize > 8*GIBIBYTE
 
-	return nil
+	var writer strings.Builder
+
+	_, _ = fmt.Fprintf(&writer, "-Xms%d\n", jvmHeapSize) // min heap size
```

Comment:
Sorry, this looks too weird. Can we just ignore the outputs here? I know we've discussed this before, but I'm pretty sure it would be fine in this case.

Also, strings.Builder has a WriteString method, so maybe we can do something like

`writer.WriteString(....)` or `writer.WriteString(fmt.Printf(...))`, in some sense it's more clear what it does.